docs/code-quality-tasks/singleton-patterns-report.md

```diff
@@ -0,0 +1,151 @@
+# Singleton Patterns Report
+
+Based on the analysis of the codebase against AGENTS.md guidelines, the
+following singleton patterns were found that should be refactored:
+
+## 1. Module-level Maps and Caches
+
+### `packages/deno-vite-plugin/src/resolver.ts`
```

Comment:
This vendored directory should probably be ignored for all of these agent requests